L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc

```diff
@@ -0,0 +1,314 @@
+#include "L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include <cmath>  // std::fabs
+
+HPSPFTauProducer::HPSPFTauProducer(const edm::ParameterSet& cfg)
+    : moduleLabel_(cfg.getParameter<std::string>("@module_label")),
+      tauBuilder_(cfg),
+      useChargedPFCandSeeds_(cfg.getParameter<bool>("useChargedPFCandSeeds")),
+      minSeedChargedPFCandPt_(cfg.getParameter<double>("minSeedChargedPFCandPt")),
+      maxSeedChargedPFCandEta_(cfg.getParameter<double>("maxSeedChargedPFCandEta")),
+      maxSeedChargedPFCandDz_(cfg.getParameter<double>("maxSeedChargedPFCandDz")),
+      useJetSeeds_(cfg.getParameter<bool>("useJetSeeds")),
+      minSeedJetPt_(cfg.getParameter<double>("minSeedJetPt")),
+      maxSeedJetEta_(cfg.getParameter<double>("maxSeedJetEta")),
+      minPFTauPt_(cfg.getParameter<double>("minPFTauPt")),
+      maxPFTauEta_(cfg.getParameter<double>("maxPFTauEta")),
+      minLeadChargedPFCandPt_(cfg.getParameter<double>("minLeadChargedPFCandPt")),
+      maxLeadChargedPFCandEta_(cfg.getParameter<double>("maxLeadChargedPFCandEta")),
+      maxLeadChargedPFCandDz_(cfg.getParameter<double>("maxLeadChargedPFCandDz")),
+      maxChargedIso_(cfg.getParameter<double>("maxChargedIso")),
+      maxChargedRelIso_(cfg.getParameter<double>("maxChargedRelIso")),
+      deltaRCleaning_(cfg.getParameter<double>("deltaRCleaning")),
+      applyPreselection_(cfg.getParameter<bool>("applyPreselection")),
+      debug_(cfg.getUntrackedParameter<bool>("debug", false)) {
+  if (debug_) {
+    std::cout << "<HPSPFTauProducer::HPSPFTauProducer (moduleLabel = " << moduleLabel_ << ")>:" << std::endl;
```

Comment:
Is this OK to use cout, instead of Log in the plugin? @silviodonato @qliphy

Reply:
Please use MessageLogger,
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger